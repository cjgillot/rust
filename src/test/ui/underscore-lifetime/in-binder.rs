// Check that we error when `'_` appears as the name of a lifetime parameter.
//
// Regression test for #52098.

struct IceCube<'a> {
    v: Vec<&'a char>,
}

impl<'_> IceCube<'_> {}
//~^ ERROR `'_` cannot be used here

struct Struct<'_> {
    //~^ ERROR `'_` cannot be used here
    //~| ERROR parameter `'_` is never used
    v: Vec<&'static char>,
}

enum Enum<'_> {
    //~^ ERROR `'_` cannot be used here
    //~| ERROR parameter `'_` is never used
    Variant,
}

union Union<'_> {
    //~^ ERROR `'_` cannot be used here
    //~| ERROR parameter `'_` is never used
    a: u32,
}

trait Trait<'_> {
    //~^ ERROR `'_` cannot be used here
}

fn foo<'_>() {
    //~^ ERROR `'_` cannot be used here
}

fn main() {}
