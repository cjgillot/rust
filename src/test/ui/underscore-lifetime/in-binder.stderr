error[E0637]: `'_` cannot be used here
  --> $DIR/in-binder.rs:9:6
   |
LL | impl<'_> IceCube<'_> {}
   |      ^^ `'_` is a reserved lifetime name

error[E0637]: `'_` cannot be used here
  --> $DIR/in-binder.rs:12:15
   |
LL | struct Struct<'_> {
   |               ^^ `'_` is a reserved lifetime name

error[E0637]: `'_` cannot be used here
  --> $DIR/in-binder.rs:18:11
   |
LL | enum Enum<'_> {
   |           ^^ `'_` is a reserved lifetime name

error[E0637]: `'_` cannot be used here
  --> $DIR/in-binder.rs:24:13
   |
LL | union Union<'_> {
   |             ^^ `'_` is a reserved lifetime name

error[E0637]: `'_` cannot be used here
  --> $DIR/in-binder.rs:30:13
   |
LL | trait Trait<'_> {
   |             ^^ `'_` is a reserved lifetime name

error[E0637]: `'_` cannot be used here
  --> $DIR/in-binder.rs:34:8
   |
LL | fn foo<'_>() {
   |        ^^ `'_` is a reserved lifetime name

error[E0392]: parameter `'_` is never used
  --> $DIR/in-binder.rs:12:15
   |
LL | struct Struct<'_> {
   |               ^^ unused parameter
   |
   = help: consider removing `'_`, referring to it in a field, or using a marker such as `PhantomData`

error[E0392]: parameter `'_` is never used
  --> $DIR/in-binder.rs:18:11
   |
LL | enum Enum<'_> {
   |           ^^ unused parameter
   |
   = help: consider removing `'_`, referring to it in a field, or using a marker such as `PhantomData`

error[E0392]: parameter `'_` is never used
  --> $DIR/in-binder.rs:24:13
   |
LL | union Union<'_> {
   |             ^^ unused parameter
   |
   = help: consider removing `'_`, referring to it in a field, or using a marker such as `PhantomData`

error: aborting due to 9 previous errors

Some errors have detailed explanations: E0392, E0637.
For more information about an error, try `rustc --explain E0392`.
